src/Common/Database/QueryUtils.php

```diff
@@ -33,6 +33,19 @@ public static function listTableFields($table)
         return $field_list;
     }
 
+    public static function escapeTableName($table)
+    {
+        return \escape_table_name($table);
+    }
+
+    public static function escapeColumnName($columnName)
+    {
+
+        // TODO: @adunsulag figure out how to fix this and make it less expensive...
```

Comment:
This is much less expensive if you also know the table (should have a second parameter). The escape tablename and escape column name are powerful prevention of sql injection since they utilize whitelisting..

Reply:
Note the 2nd parameter in escape_sql_column_name() function is an array of table names

Reply:
I used this.  Its still not super performant as it hits a list of columns for the table.  Making this change looks like it adds quite the performance hit (it about doubled my query time) but I guess that's safer than trusting the xml files.  A lot more could be done to optimize all of this but we'll have to see if we can get to it once documentation is done.

Reply:
I figured out a different optimization and moved the escaping to another location.